geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/SimpleFeatureValidator.scala

```diff
@@ -8,168 +8,256 @@
-      lastErr = null
-      i = sft.getGeomIndex
+  object HasGeoValidator extends ValidatorFactory {
+    override val name: String = "has-geo"
+    override def validator(sft: SimpleFeatureType, config: Option[String]): Validator = {
+      val i = sft.getGeomIndex
+      if (i == -1) { NoValidator } else { new NullValidator(i, Errors.GeomNull) }
     }
+  }
 
-    override def validate(sf: SimpleFeature): Boolean = {
-      if (i == -1) { true } else {
-        val res = sf.getAttribute(i) != null
-        lastErr = if (res) { null } else { "Null geometry attribute" }
-        res
-      }
+  object HasDtgValidator extends ValidatorFactory {
+    override val name: String = "has-dtg"
+    override def validator(sft: SimpleFeatureType, config: Option[String]): Validator = {
+      val i = sft.getDtgIndex.getOrElse(-1)
+      if (i == -1) { NoValidator } else { new NullValidator(i, Errors.DateNull) }
     }
-
-    override def lastError: String = lastErr
   }
 
-  object ZIndexValidator {
-    val name: String = "z-index"
+  object NoneValidator extends ValidatorFactory {
+    override val name: String = "none"
+    override def validator(sft: SimpleFeatureType, config: Option[String]): Validator = NoValidator
   }
 
   /**
-    * Validates that the geometry fits into a Z2 or Z3 style WGS84 index scheme
+    * Validates an attribute is not null
+    *
+    * @param i attribute index
+    * @param error error message
     */
-  class ZIndexValidator extends SimpleFeatureValidator {
-
-    import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
-    import org.locationtech.geomesa.utils.geotools.wholeWorldEnvelope
-
-    private var dtgIndex: Int = -1
-    private var geomIndex: Int = -1
-    private var minDate: Date = _
-    private var maxDate: Date = _
-    private var lastErr: String = _
-
-    override def name: String = ZIndexValidator.name
-
-    override def init(sft: SimpleFeatureType): Unit = {
-      lastErr = null
-      dtgIndex = sft.getDtgIndex.getOrElse(-1)
-      geomIndex = sft.getGeomIndex
-      minDate = Date.from(BinnedTime.ZMinDate.toInstant)
-      maxDate = Date.from(BinnedTime.maxDate(sft.getZ3Interval).toInstant)
+  private class NullValidator(i: Int, error: String) extends Validator {
+    override def validate(sf: SimpleFeature): String = if (sf.getAttribute(i) == null) { error } else { null}
+  }
 
-      if (dtgIndex == -1 || geomIndex == -1) {
-        throw new IllegalArgumentException("Z3 validator cannot be used on a type lacking a date or geometry")
+  /**
+    * Z2 validator
+    *
+    * @param geom geom index
+    */
+  private class Z2Validator(geom: Int) extends Validator {
+    override def validate(sf: SimpleFeature): String = {
+      val g = sf.getAttribute(geom).asInstanceOf[Geometry]
+      if (g == null) {
+        Errors.GeomNull
+      } else if (!wholeWorldEnvelope.contains(g.getEnvelopeInternal)) {
+        Errors.GeomBounds
```

Comment:
As a follow-on, how about an XZ2 validator which only requires that the envelope/geometry intersects with the wholeWorldEnvelope?